src/clients/Deluge.ts

```diff
@@ -0,0 +1,228 @@
+	}
+
+	/**
+	 * checks enabled plugins for "Label"
+	 * returns true if successful.
+	 */
+	private async labelEnabled() {
+		const enabledLabels = await this.call("core.get_enabled_plugins", []);
+		return enabledLabels?.result?.includes?.("Label");
+	}
+
+	/**
+	 * if Label plugin is loaded, adds (if necessary)
+	 * and sets the label based on torrent hash.
+	 */
+	private async setLabel(infoHash: string, label: string): Promise<void> {
+		if (this.isLabelEnabled) {
+			const setResult = await this.call("label.set_torrent", [
+				infoHash,
+				label,
+			]);
+			if (setResult?.error?.code == DelugeErrorCode.RPC_FAIL) {
+				await this.call("label.add", [label]);
+				await this.call("label.set_torrent", [infoHash, label]);
+			}
+		}
+	}
+
+	/**
+	 * injects a torrent into deluge client
+	 */
+	async inject(
+		newTorrent: Metafile,
+		searchee: Searchee,
+		path?: string
+	): Promise<InjectionResult> {
+		let torrentInfo;
+		if (searchee.infoHash) {
+			torrentInfo = await this.getTorrentInfo(searchee);
+			if (!torrentInfo.complete) {
+				return InjectionResult.TORRENT_NOT_COMPLETE;
+			}
+		}
+
+		const params = this.formatData(
+			`${newTorrent.name}.cross-seed.torrent`,
+			newTorrent.encode().toString("base64"),
+			path ? path : torrentInfo.save_path
+		);
+		const addResult = (await this.call(
+			"core.add_torrent_file",
+			params
+		)) as DelugeResponse;
+		if (addResult?.result) {
+			if (searchee.infoHash) {
+				await this.setLabel(newTorrent.infoHash, this.delugeLabel);
+			} else {
+				const { dataCategory } = getRuntimeConfig();
+				await this.setLabel(newTorrent.infoHash, dataCategory);
+			}
+			return InjectionResult.SUCCESS;
+		} else if (addResult?.error?.message?.includes("already")) {
+			return InjectionResult.ALREADY_EXISTS;
+		} else if (addResult?.error?.message) {
+			logger.debug({
+				label: Label.DELUGE,
+				message: `injection failed: ${addResult.error.message}`,
+			});
+			return InjectionResult.FAILURE;
+		}
+	}
+
+	/**
+	 * formats the json for rpc calls to inject
+	 */
+	private formatData(filename: string, filedump: string, path: string) {
+		return [
+			filename,
+			filedump,
+			{
+				add_paused: false,
+				seed_mode: getRuntimeConfig().skipRecheck,
+				download_location: path,
+			},
+		];
+	}
+
+	/**
+	 * returns information needed to complete/validate injection
+	 */
+	private async getTorrentInfo(searchee: Searchee): Promise<TorrentInfo> {
+		try {
+			const params = [
+				["state", "progress", "save_path"],
+				{ hash: searchee.infoHash },
+			];
+
+			const response = await this.call("web.update_ui", params);
+			const completedTorrent =
+				response?.result?.torrents?.[searchee.infoHash]?.state ===
+					"Seeding" ||
+				response?.result?.torrents?.[searchee.infoHash]?.progress ===
+					100;
+			return {
+				complete: completedTorrent,
+				save_path: completedTorrent
+					? response?.result?.torrents?.[searchee.infoHash]?.save_path
+					: undefined,
+			} as TorrentInfo;
```

Comment:
Typescript should be able to infer that this is a TorrentInfo since we're creating the object ourselves

```suggestion
			};
```